src/Indicator.vala

```diff
@@ -27,14 +27,41 @@ public class Power.Indicator : Wingpanel.Indicator {
     private Services.Device primary_battery;
     private bool notify_battery = false;
 
+    private Widgets.ScreenBrightness screen_brightness;
+
     public Indicator (bool is_in_session) {
         Object (code_name : Wingpanel.Indicator.POWER,
                 display_name : _("Power"),
                 description: _("Power indicator"));
 
-        this.is_in_session = is_in_session;
+        display_widget = new Widgets.DisplayWidget ();
+        screen_brightness = new Widgets.ScreenBrightness ();
+        popover_widget = new Widgets.PopoverWidget (screen_brightness, is_in_session);
+
+        popover_widget.settings_shown.connect (() => close ());
+
+        var dm = Services.DeviceManager.get_default ();
+
+        /* No need to display the indicator when the device is completely in AC mode */
+        if (dm.has_battery || dm.backlight.present) {
+            update_visibility ();
+        }
+        dm.notify["has-battery"].connect (update_visibility);        
+
+        display_widget.scroll_event.connect (on_icon_scroll_event);
     }
 
+    construct { }
+
+    private bool on_icon_scroll_event (Gdk.EventScroll e) {
+            if (e.direction == Gdk.ScrollDirection.UP) {
+                screen_brightness.set_value(screen_brightness.get_value () + 10);
```

Comment:
I don't know if the increment should be a fixed amount. If `e.scroll_direction == Clutter.ScrollDirection.SMOOTH` I think it should be a function of the `e.x` value. And perhaps later we should do the same with using scroll on the slider to.

Reply:
Also have doubts about this, but I'm not familiar with Clutter. Maybe someone else can do this?